vispy/visuals/markers.py

```diff
@@ -617,7 +618,7 @@ def set_data(self, pos=None, size=10., edge_width=1., edge_width_rel=None,
                 raise ValueError('edge_width_rel cannot be negative')
 
         if symbol is not None:
-            if not np.all(np.isin(np.asarray(symbol), self.symbols)):
+            if not all(x in self._symbol_shader_values for x in set(symbol if isinstance(symbol, Iterable) else [symbol])):
```

Comment:
I think this will be a problem if someone passes `symbol="disc"` as strings are `Iterable`.
```suggestion
            if not all(x in self._symbol_shader_values for x in set(symbol if isinstance(symbol, Iterable) and not isinstance(symbol, str) else [symbol])):
```